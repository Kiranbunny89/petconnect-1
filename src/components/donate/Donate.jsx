import "./donate.css";
import charity from "./charities.json";
import { Card, Image, Button } from "react-bootstrap";

const Donate = () => {
  return (
    <div className="donate">
      <h1>DONATE</h1>
      <section className="kofi__section">
        <p>
          Here are a list of places to donate to help pets and animals. Feel
          free to suggest any more charities. Could also message me if you
          donated and I'll add to a list of donors
        </p>
        <p>If you wish to help donate the site here, you can click here</p>
        <a href={process.env.REACT_APP_KOFI} target="_blank" rel="noreferrer">
          <Image
            src="https://cdn.ko-fi.com/cdn/kofi2.png?v=2"
            className="kofi__image"
            alt="Buy Me a Coffee at ko-fi.com"
          />
        </a>
      </section>
      <section className="charity__section">
        {charity.map((ch) => (
          <Card key={ch.name}>
            <Card.Img variant="top" src={ch.logo} className="" />
            <Card.Body>
              <Card.Title>{ch.name}</Card.Title>
              <Card.Subtitle className="mb-2 text-muted">
                Location: {ch.location}
              </Card.Subtitle>
              <Card.Subtitle className="mb-2 text-muted">
                Founded: {ch.founded}
              </Card.Subtitle>
              <Card.Text>{ch.mission}</Card.Text>
              <Button variant="primary" href={ch.website} target="_blank">
                Click Here
              </Button>
            </Card.Body>
          </Card>
        ))}
      </section>
    </div>
  );
};
export default Donate;
