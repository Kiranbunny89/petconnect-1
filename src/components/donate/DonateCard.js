import { Card, Button } from "react-bootstrap";

const DonateCard = ({ ch }) => (
  <Card>
    <Card.Img variant="top" src={ch.logo} />
    <Card.Body>
      <Card.Title>{ch.name}</Card.Title>
      <Card.Subtitle className="mb-2 text-muted">
        Location: {ch.location}
      </Card.Subtitle>
      <Card.Subtitle className="mb-2 text-muted">
        Founded: {ch.founded}
      </Card.Subtitle>
      <Card.Text>{ch.mission}</Card.Text>
      <Button variant="primary" href={ch.website} target="_blank">
        Click Here
      </Button>
    </Card.Body>
  </Card>
);
export default DonateCard;
