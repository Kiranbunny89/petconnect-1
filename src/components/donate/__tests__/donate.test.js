import { cleanup, render, screen } from "@testing-library/react";
import Donate from "../Donate";

import renderer from "react-test-renderer";
import DonateCard from "../DonateCard";
afterEach(() => {
  cleanup();
});

test("should render donate component", () => {
  render(<Donate />);
  const donateElement = screen.getByTestId("donate-test-title");
  expect(donateElement).toBeInTheDocument();

  expect(donateElement).toHaveTextContent("DONATE");
});

test("matches donate card snapshot", () => {
  const sampleData = {
    name: "Lorem Charity",
    website: "https://via.placeholder.com/",
    logo: "https://via.placeholder.com/150",
    location: "Lorem States",
    founded: 9000,
    mission: "Peace of Lorem",
  };
  const tree = renderer.create(<DonateCard ch={sampleData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
