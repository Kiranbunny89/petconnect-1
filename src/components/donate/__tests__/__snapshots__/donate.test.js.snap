// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches donate card snapshot 1`] = `
<div
  className="card"
>
  <div
    className="row"
  >
    <div
      className="m-auto col-sm-4"
    >
      <img
        className="card-img-left d-block mx-auto img-fluid w-50"
        src="https://via.placeholder.com/150"
      />
    </div>
    <div
      className="col-sm-8"
    >
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          Lorem Charity
        </div>
        <div
          className="mb-2 text-muted card-subtitle h6"
        >
          Location: 
          Lorem States
        </div>
        <div
          className="mb-2 text-muted card-subtitle h6"
        >
          Founded: 
          9000
        </div>
        <p
          className="card-text"
        >
          Peace of Lorem
        </p>
        <a
          className="btn btn-primary"
          disabled={false}
          href="https://via.placeholder.com/"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
          target="_blank"
        >
          Click Here
        </a>
      </div>
    </div>
  </div>
</div>
`;
